src/test/java/org/alfresco/repo/content/replication/AggregatingContentStoreTest.java

```diff
@@ -23,145 +23,220 @@
+     * 
+     * @param contentUrl String
+     * @param mustExist true if the content must exist, false if it must <b>not</b> exist
+     */
+    private void checkForUrl(String contentUrl, boolean mustExist)
     {
         ContentReader reader = getReader(contentUrl);
-        assertEquals("Reader state differs from expected: " + reader, mustExist, reader.exists());
-    }
-    
-    public void testDelete() throws Exception
-    {
-        
-        // write some content
-        ContentWriter writer = getWriter();
-        writer.putContent(SOME_CONTENT);
-        String contentUrl = writer.getContentUrl();
-        
-        ContentReader reader = primaryStore.getReader(contentUrl);
-        assertTrue("Content was not in the primary store", reader.exists());
-        assertEquals("The content was incorrect", SOME_CONTENT, reader.getContentString());
-
-        getStore().delete(contentUrl);
-        checkForUrl(contentUrl, false);
-    }
-    
-    public void testReadFromSecondaryStore()
-    {
-        // pick a secondary store and write some content to it
-        ContentStore secondaryStore = secondaryStores.get(2);
-        ContentWriter writer = secondaryStore.getWriter(ContentContext.NULL_CONTEXT);
-        writer.putContent(SOME_CONTENT);
-        String contentUrl = writer.getContentUrl();
-        
-        checkForUrl(contentUrl, true);
-    }
-  
-
-}
+        assertEquals("Reader state differs from expected: " + reader, mustExist, reader.exists());
+    }
+    
+    public void testDelete() throws Exception
+    {
+        
+        // write some content
+        ContentWriter writer = getWriter();
+        writer.putContent(SOME_CONTENT);
+        String contentUrl = writer.getContentUrl();
+        
+        ContentReader reader = primaryStore.getReader(contentUrl);
+        assertTrue("Content was not in the primary store", reader.exists());
+        assertEquals("The content was incorrect", SOME_CONTENT, reader.getContentString());
+
+        getStore().delete(contentUrl);
+        checkForUrl(contentUrl, false);
+    }
+    
+    public void testReadFromSecondaryStore()
+    {
+        // pick a secondary store and write some content to it
+        ContentStore secondaryStore = secondaryStores.get(2);
+        ContentWriter writer = secondaryStore.getWriter(ContentContext.NULL_CONTEXT);
+        writer.putContent(SOME_CONTENT);
+        String contentUrl = writer.getContentUrl();
+        
+        checkForUrl(contentUrl, true);
+    }
+
+    public void testIsDirectAccessSupported()
+    {
+        // Create the aggregating store
+        AggregatingContentStore aggStore = new AggregatingContentStore();
+        aggStore.setPrimaryStore(primaryStoreMock);
+        aggStore.setSecondaryStores(List.of(secondaryStoreMock));
+
+        // By default it is unsupported
+        assertFalse(aggStore.isDirectAccessSupported());
+        
+        // Not supported if the primary store doesn't support it and the secondary do.
```

Comment:
I looks in the code that it would go and check in the secondary store even when not supported by primary one.

Reply:
My bad, indeed `isDirectAccessSupported` checks primary store. But if you call `getDirectAccessUrl` is still checking secondary stores, when primary doesn't support it.

Reply:
I updated this, thank you